daemon/src/monitor.rs

```diff
@@ -117,19 +117,29 @@ pub struct Actor {
 struct Cfd {
     id: OrderId,
 
-    monitor_lock_finality: Option<Lock>,
-    monitor_commit_finality: Option<Commit>,
+    lock: Option<Lock>,
+    monitor_lock_finality: bool,
+
+    collaborative_settlement: Option<(Txid, Script)>,
+    monitor_collaborative_settlement_finality: bool,
+
+    commit: Option<Commit>,
+    monitor_commit_finality: bool,
     monitor_cet_timelock: bool,
     monitor_refund_timelock: bool,
-    monitor_refund_finality: Option<Refund>,
+
+    cet: Option<(Txid, Script)>,
+    monitor_cet_finality: bool,
+
+    refund: Option<Refund>,
+    monitor_refund_finality: bool,
+
     monitor_revoked_commit_transactions: Vec<RevokedCommit>,
-    monitor_collaborative_settlement_finality: Option<(Txid, Script)>,
-    monitor_cet_finality: Option<(Txid, Script)>,
 
     // Rebroadcast transactions upon startup
-    lock_tx: Option<Transaction>,
-    cet: Option<Transaction>,
-    commit_tx: Option<Transaction>,
+    broadcast_lock: Option<Transaction>,
+    broadcast_cet: Option<Transaction>,
+    broadcast_commit: Option<Transaction>,
```

Comment:
🤔 Unless I missed something `broadcast_commit` is never set to `Some`. Is that on purpose or is there something missing here?

Reply:
Good catch! But looking at the diff we never actually set it to `Some` before either! Which means if we ever found ourselves in a situation where a restart occurred after a `ManualCommit` was emitted but the commit transaction publication triggered by the `process_manager` _failed_, we would have been stuck.

I'll add another commit to fix this.

Reply:
Done with https://github.com/itchysats/itchysats/pull/3029/commits/63f0a89c801326e5fe3f045efb32d17e3a3c6d95.